File: 1. Two Sum/TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let left = 0;
    let right = nums.length-1;
    let newArray = Array.from(nums);
    nums.sort((a,b)=> {return a-b});

    while(left < right)
    {   
        let total = nums[left] + nums[right];
        if (total == target) {
            break;
        }
        else
        {
           if (total < target) {
                left++;
           }
           else
           {
            right--;
           }
        }
    }
    return [newArray.indexOf(nums[left]), newArray.lastIndexOf(nums[right])]
};

File: 1. Two Sum/TwoSum.cpp
class Solution {
    public:
        vector<int> twoSum(vector<int>& nums, int target) {
            unordered_map<int, int> myMap;
            for (int i = 0; i < nums.size(); i++)
            {
                int temp = nums[i] ;
                if (myMap.contains(temp))
                {
                    return {myMap[temp], i}; 
                }
                myMap[target -temp] = i;   
            }
            return {};
        }
    };
    
    
    